perfkitbenchmarker/linux_benchmarks/aws_dynamodb_ycsb_benchmark.py

```diff
@@ -0,0 +1,162 @@
+The only VM group is client group that sends requests to specifiedDB.
+TODO: add RANGE option.
+TODO: add DAX option.
+TODO: add global table option.
+"""
+
+import logging
+import posixpath
+from perfkitbenchmarker import configs
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import vm_util
+from perfkitbenchmarker.linux_packages import ycsb
+from perfkitbenchmarker.providers.aws import aws_dynamodb
+
+BENCHMARK_NAME = 'aws_dynamodb_ycsb'
+BENCHMARK_CONFIG = """
+aws_dynamodb_ycsb:
+  description: >
+      Run YCSB against AWS DynamoDB.
+      Configure the number of VMs via --ycsb_client_vms.
+  vm_groups:
+    default:
+      vm_spec: *default_single_core
+      vm_count: 1"""
+
+YCSB_BINDING_LIB_DIR = posixpath.join(ycsb.YCSB_DIR, 'dynamodb-binding', 'lib')
+AWS_CREDENTIAL_DIR = '/tmp/AWSCredentials.properties'
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string('aws_dynamodb_ycsb_awscredentials_properties',
+                    './AWSCredentials.properties',
+                    'The AWS credential location. Defaults to PKB top folder')
+flags.DEFINE_string('aws_dynamodb_ycsb_dynamodb_primarykey',
+                    'primary_key',
+                    'The primaryKey of dynamodb table.')
+flags.DEFINE_string('aws_dynamodb_ycsb_dynamodb_region',
+                    None,
+                    'The AWS dynamodb region to connect to.'
+                    'Default is to use zones.')
+flags.DEFINE_string('aws_dynamodb_ycsb_readproportion',
+                    '0.5',
+                    'The read proportion, '
+                    'default is 0.5 in workloada and 0.95 in YCSB.')
+flags.DEFINE_string('aws_dynamodb_ycsb_updateproportion',
+                    '0.5',
+                    'The update proportion, '
+                    'default is 0.5 in workloada and 0.05 in YCSB')
+flags.DEFINE_string('aws_dynamodb_ycsb_table',
+                    'pkb',
+                    'The dynamodb table name precursor.')
+flags.DEFINE_enum('aws_dynamodb_ycsb_consistentReads',
+                  None, ['false', 'true'],
+                  "Consistent reads cost 2x eventual reads. "
+                  "'false' is default which is eventual")
+
+
+def GetConfig(user_config):
+    config = configs.LoadConfig(BENCHMARK_CONFIG, user_config, BENCHMARK_NAME)
+    if FLAGS['ycsb_client_vms'].present:
+        config['vm_groups']['default']['vm_count'] = FLAGS.ycsb_client_vms
+    return config
+
+
+def CheckPrerequisites(benchmark_config):
+    """Verifies that the required resources are present.
+    Raises:
+    perfkitbenchmarker.data.ResourceNotFound: On missing resource.
+    """
+    ycsb.CheckPrerequisites()
+
+
+def Prepare(benchmark_spec):
+    """Install YCSB on the target vm.
+    Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+    """
+    if FLAGS.aws_dynamodb_ycsb_dynamodb_region is None:
+      prereg = FLAGS.zones[0]
```

Comment:
zone is different from region?
zone is something like us-east1-a
region is us-east1?

Reply:
region is like 'us-east-1' and availability zones are like 'us-east-1c'  it looked like from what i was seeing that 'zones' were being treated like 'regions' since that what aws_virtual_machine was using, and it was grabbing that from the 'default' aws region.